pytorch_lightning/trainer/trainer.py

```diff
@@ -387,7 +387,7 @@ def __init__(
         gpu_ids, tpu_cores = self._parse_devices(gpus, auto_select_gpus, tpu_cores)
 
         # init connectors
-        self.config_validator = ConfigValidator(self)
+        self._config_validator = ConfigValidator(self)
```

Comment:
this doesn't need to take the trainer as input in the constructor at all. config validator can be recast as a utility method which accepts a trainer and lightning module.

that can be a followup to this PR